internal/pkg/agent/install/uninstall.go

```diff
@@ -41,23 +44,28 @@ func Uninstall(cfgFile, topPath, uninstallToken string) error {
 	if status == service.StatusRunning {
 		err := svc.Stop()
 		if err != nil {
-			return errors.New(
+			return aerrors.New(
 				err,
 				fmt.Sprintf("failed to stop service (%s)", paths.ServiceName),
-				errors.M("service", paths.ServiceName))
+				aerrors.M("service", paths.ServiceName))
 		}
 	}
 
+	// kill any running watcher
+	if err := killWatcher(); err != nil {
+		return fmt.Errorf("failed trying to kill any running watcher: %w", err)
```

Comment:
Does this need to be a fatal error? Maybe this makes sense behind a `--force` flag as discussed in https://github.com/elastic/elastic-agent/pull/3373#discussion_r1319119781

Reply:
This step is right after the service is stopped, but not removed. Being that it is early I prefer that we ensure that killing the watcher works, before going any futher. This is not leaving the Elastic Agent in a broken state, the service is just off.

We could maybe add skip the error in the case that `--force` is added, but don't want to conflate this PR with an addition of `--force`.